Env: Update README.md

MikeiLL opened the pull request:
It took me a bit of hunting to find what might be a fairly common use case. Perhaps this will be useful.

## What?
Providing an example for config to update URI/URL for dev site.

## Why?
Ease of understanding and use.

## How?
Just adding a little example.

## Testing Instructions
N/A

MikeiLL committed:
Update README.md

It took me a bit of hunting to find what might be a fairly common use case. Perhaps this will be useful.

noahtallen commented on packages/env/README.md:
```diff
@@ -507,6 +507,16 @@ On the development instance, `cwd` will be mapped as a plugin, `one-theme` will
 
 On the tests instance, `cwd` is still mapped as a plugin, but no theme is mapped. Additionally, while KEY_2 is still set to false, KEY_1 is overridden and set to false. 3000 overrides the default port as well.
 
+Another example would be serving the development site over a different URL, like this:
```
Could you move this to the "examples" section somewhere below line 547 and include a heading? That should keep the file a bit more consistent :)

MikeiLL replied:
How did I miss that there was an "examples" section!? Thank you.

MikeiLL replied:
Something I'd love to understand better and possibly note in the docs is _how_ to look and/or _where_ to find keys (`DOCKER_REQUEST_URL` for example) that can be used for various goals in the env.

noahtallen replied:
So the `config` option is just letting you write a PHP constant to `wp-config.php`. So the constants are whatever WordPress allows (e.g. https://wpengineer.com/2382/wordpress-constants-overview/)

`DOCKER_REQUEST_URL` is not one of those constants, though -- as far as I can tell, it was only ever suggested here (#23947) as a way to enable this feature

MikeiLL replied:
> So my big question with this PR -- does this actually work? It does not work for me locally, and as far as I can tell the issue requesting the feature (#23947) hasn't been implemented (yet). 🤔

It works for me. I'm serving a site over `mzmbo.test:8888` I'm interested in #23947 because I want to test some Oauth features which require an `https` callback url. In terms of the domain name, though, as I say, working for me. I don't know, yet, if serving over port `80` would make it so that I don't need to include the port in the browser url.

noahtallen replied:
did you have to set anything up in your /etc/hosts file?

noahtallen reviewed (commented):
Thanks for adding this! Just one small comment to resolve and than I can merge it :)

MikeiLL committed:
Update README.md

Migrate specified url example to examples section and add header as per PR comment.

noahtallen commented:
So my big question with this PR -- does this actually work? It does not work for me locally, and as far as I can tell the issue requesting the feature (https://github.com/WordPress/gutenberg/issues/23947) hasn't been implemented (yet). 🤔

MikeiLL commented:
Oh. Duh. Yes. That's embarrassing. 

I have this in there from a previous (Docker-based) approach: `127.0.0.1 localhost mzmbo.test`

I could add a line about that to the README in the PR.

> On Oct 11, 2022, at 2:48 PM, Noah Allen ***@***.***> wrote:
> 
> 
> @noahtallen commented on this pull request.
> 
> In packages/env/README.md <https://github.com/WordPress/gutenberg/pull/44836#discussion_r992718905>:
> 
> > @@ -507,6 +507,16 @@ On the development instance, `cwd` will be mapped as a plugin, `one-theme` will
>  
>  On the tests instance, `cwd` is still mapped as a plugin, but no theme is mapped. Additionally, while KEY_2 is still set to false, KEY_1 is overridden and set to false. 3000 overrides the default port as well.
>  
> +Another example would be serving the development site over a different URL, like this:
> did you have to set anything up in your /etc/hosts file?
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/WordPress/gutenberg/pull/44836#discussion_r992718905>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABQYTR2RN647HHJQEDZTUXLWCXAA7ANCNFSM6AAAAAARBUUYMY>.
> You are receiving this because you were mentioned.
>

noahtallen commented:
Ahh, that makes sense! I think we should wait to merge this until #23947 is added, just because wp-env isn't technically supporting this use case and I don't know if there are any side effects.

MikeiLL commented:
Makes sense to me. 

> On Oct 12, 2022, at 1:04 PM, Noah Allen ***@***.***> wrote:
> 
> 
> Ahh, that makes sense! I think we should wait to merge this until #23947 <https://github.com/WordPress/gutenberg/issues/23947> is added, just because wp-env isn't technically supporting this use case and I don't know if there are any side effects.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/WordPress/gutenberg/pull/44836#issuecomment-1276548058>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABQYTR2MBLYSSPMBZNUK7FDWC34TZANCNFSM6AAAAAARBUUYMY>.
> You are receiving this because you were mentioned.
>
